src/BackpackServiceProvider.php

```diff
@@ -83,6 +83,19 @@ public function registerMiddlewareGroup(Router $router)
         $middleware_key = config('backpack.base.middleware_key');
         $middleware_class = config('backpack.base.middleware_class');
 
+        //we get the web middleware defined in config, could be a single key: web for example,
+        //or it can be an array of middlewares that replaces the default `web` middleware from Laravel.
+        $web_middleware = config('backpack.base.web_middleware', 'web');
+
+        //if it is an array it means developer would like to replace the default web middleware list with his own MW array.
+        if (is_array($web_middleware)) {
+            $router->middlewareGroup('web', $web_middleware);
```

Comment:
Hmm.. Pedro are you sure about this? If this is in the backpack service provider, doesn't this mean that this would effectively _replace_ the web middleware, in all pages? Even non-backpack pages?

Reply:
I can reproduce the error now @pxpm , but I'm still skeptical about this implementation.
Are you sure this doesn't replace the `web` middleware with the one in the Backpack config for ALL pages, not just Backpack pages?

Reply:
If this DOES extend the scope of the feature, like I think, an alternative fix would probably be inside the route files themselves. I'm wondering if using `array_merge()` would't just fix this entirely:

```diff
Route::group([
    'prefix'     => config('backpack.base.route_prefix', 'admin'),
-    'middleware' => [
-        config('backpack.base.web_middleware', 'web'), 
-        config('backpack.base.middleware_key', 'admin')
-    ],
+    'middleware' => array_merge(
+        config('backpack.base.web_middleware', 'web'), 
+        config('backpack.base.middleware_key', 'admin')
+   ),
    'namespace'  => 'App\Http\Controllers\Admin',
], function () { // custom admin routes
```

This would have the _downside_ of being in the route file, so people won't get the fix _automatically_. But it would have the _upside_ of being in the route file 😂 so its scope is limited, and it would be more transparent (we're not changing a config value the developer gave with something else - that could be confusing).

Your thoughts @pxpm ?

Reply:
Indeed @tabacitu this would be out of scope. I am sorry, I changed in route files but find that it would be a breaking change like you described, so I went to change in Service Provider. I missed the `this is not a replacement for web middleware`. 

From my understanding I think the way we do it is not intuitive. The setting is called `web_middleware`, so it makes sense that you would overwrite `web` middleware group if you change it. 

I think it would be better to use something like: `middleware => '['admin', 'web']'` and if people don't want to use the default web middleware group they should change it in config to: `middleware => ['admin', 'whatever_middleware::class', 'whatever_middleware_two::class]` instead of the way we currrently do it that I think it's not very clear.

This **is** a bug indeed, not a possible one. I am going to test your proposed solution, mine should have that downside, I didn't notice it, all my panels does not mix both middlewares, I push middlewares in `web` group into `admin` and only use `admin` for the backpack routes. 

Let me know what you think.

Best,
Pedro